src/components/User/UserInteract.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {useState, Dispatch, SetStateAction, Fragment} from 'react';
+import './UserInteract.css';
+
+import Button from '@material-ui/core/Button';
+import AddIcon from '@material-ui/icons/Add';
+import Input from '@material-ui/core/Input';
+
+interface UserInfo {
+    firstName: string;
+    familyName: string;
+    fields: Array<string>;
+}
+
+interface UserInfoItem {
+    0: string;
+    1: string;
+    2: Array<string>;
+}
+
+export function UserInteract({contract, setSpinner}: {
+    contract: null, setSpinner: Dispatch<SetStateAction<boolean>>
+}) {
+    const [userInfos, setUserInfos] = useState<UserInfo>({firstName: '', familyName: '', fields: []});
+    const [field, setField] = useState('');
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [user, setUser] = useState<UserInfoItem | null>(null);
+
+    function handleClickRegisterUser() {
+        setUserInfos({firstName: '', familyName: '', fields: []});
+    }
+
+    function handleClickGetUser() {
+        
+    }
+
+    function handleClickAddField() {
+        setUserInfos({...userInfos, fields: [...userInfos.fields, field]});
+        setField('');
+    }
+
+    return (
+        <Fragment>
```

Comment:
Tip for the next time you write some React: using empty html tags is equivalent to using `React.Fragment`. So `<></>` is the same as `<Fragment></Fragment>`.